### PR TITLE
Optimize query platform

### PR DESCRIPTION
**Pull request by pacospace**
## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/storages/issues/2181

## This introduces a breaking change

- [ ] Yes
- [x] No

## This should yield a new module release

- [x] Yes
- [ ] No

## This Pull Request implements

- Introduce and migation for db
- Adjust queries and sync logic

### CODE REVIEW

File: thoth/storages/graph/postgres.py [hunk 1]
```diff
@@ -2293,17 +2294,24 @@ def get_python_package_version_dependents_all(
 
             return result
 
-    def python_package_version_depends_on_platform_exists(self, platform: str) -> bool:
+    @staticmethod
+    def python_package_version_depends_on_platform_exists(platform: str) -> bool:
         """Check if the given platform has some records in the database."""
-        with self._session_scope() as session:
-            return session.query(exists().where(DependsOn.platform == platform)).scalar()
+        if platform in [p.value for p in PlatformEnum]:
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:2300]:
```suggestion
    return platform in (p.value for p in PlatformEnum)\]
```

File: thoth/storages/data/alembic/versions/fd56b28d89e6_create_enum_for_platform.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+               type_=postgresql.ENUM('LINUX_X86_64', name='platform'),
+               existing_nullable=False,
+               existing_server_default=sa.text("'linux-x86_64'::text"))
+    op.alter_column('has_symbol', 'software_environment_id',
+               existing_type=sa.INTEGER(),
+               nullable=False)
+    op.alter_column('has_symbol', 'versioned_symbol_id',
+               existing_type=sa.INTEGER(),
+               nullable=False)
+    op.create_unique_constraint(None, 'python_package_version_entity_rule', ['id'])
```
Comment (fridex) [inline thoth/storages/data/alembic/versions/fd56b28d89e6_create_enum_for_platform.py:32]: These three do not correspond to the PR. Might be some leftovers not captured in older PRs.

**Review by fridex — COMMENTED** [review]
lgtm, few comments left

File: thoth/storages/data/alembic/versions/fd56b28d89e6_create_enum_for_platform.py (hunk 2 above)
Comment (pacospace) [reply thoth/storages/data/alembic/versions/fd56b28d89e6_create_enum_for_platform.py:32]: Yes not sure why they appeared, I will update after I fix usual problems with using enums that are not automatically handled by migrations

**Review by harshad16 — COMMENTED** [review]
/lgtm

**Review by fridex — APPROVED** [review]
thanks! 👍🏻

**Pull request merged**